src/model/documentselection.js

```diff
@@ -276,12 +261,17 @@ export default class DocumentSelection {
 	 *		</blockB>
 	 *		<blockE></blockE>]
 	 *
-	 * **Note:** To get all selected blocks use {@link #getSelectedBlocks `getSelectedBlocks()`}.
+	 * **Special case**: If a selection ends at the beginning of a block, that block is not returned as from user perspective
+	 * this block wasn't selected. See [#984](https://github.com/ckeditor/ckeditor5-engine/issues/984) for more details.
+	 *
+	 *		<paragraph>[a</paragraph>
+	 *		<paragraph>b</paragraph>
+	 *		<paragraph>]c</paragraph> // this block will not be returned
 	 *
 	 * @returns {Iterable.<module:engine/model/element~Element>}
 	 */
-	getTopMostBlocks() {
-		return this._selection.getTopMostBlocks();
+	getSelectedBlocks( config ) {
```

Comment:
Mmmm, do we need `config` here? It is not described in docs (or I can't see it).

Reply:
Good catch - it's a left-over. I didn't implement the `config.deep` after checking how blockquote is treated with blocks.

Reply:
Okay, AFAICS `Selection#getSelectedBlocks` does not take any arguments so this is probably an artifact after the previous solution.